### PR TITLE
test: add installation component for new e2e test of Kind feature

### PR DESCRIPTION
**Pull request by amisskii**
Signed-off-by: Anton Misskii <amisskii@redhat.com>

### What does this PR do?
This PR introduces the installation component of a new end-to-end test for the Kind feature, which was recently added to the Podman Desktop Application.

### What issues does this PR fix or reference?
#5976 #7977 

### How to test this PR?

Run the tests with yarn test:e2e:run

### CODE REVIEW

File: tests/playwright/src/model/pages/resources-page.ts [hunk 1]
```diff
@@ -25,12 +25,18 @@ export class ResourcesPage extends SettingsPage {
...
   constructor(page: Page) {
     super(page, 'Resources');
     this.heading = this.header.getByRole('heading', { name: 'Title' }).and(this.header.getByText('Resources'));
     this.featuredProviderResources = this.content.getByRole('region', { name: 'Featured Provider Resources' });
     this.podmanResources = this.featuredProviderResources.getByRole('region', { name: 'podman', exact: true });
     this.composeResources = this.featuredProviderResources.getByRole('region', { name: 'Compose', exact: true });
+    this.resource = this.page.getByRole('region', { name: 'Resources' });
+  }
+
+  public ResourceExists(label: string): Locator {
```
Comment (cbr7) [inline tests/playwright/src/model/pages/resources-page.ts]: methods should not be named starting with a capital letter and also this naming implies the methods should return a boolean value with value true with the resource Exists.

File: tests/playwright/src/model/workbench/status-bar.ts [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    this.dialogOkButton = this.kindDialogWindow.getByRole('button', { name: 'OK' });
+    this.dialogCloseButton = this.kindDialogWindow.getByRole('button', { name: 'Close' });
+  }
+
+  public async installKindCLI(): Promise<void> {
+    await this.kindInstallationButton.click({ timeout: 5000 });
+    await expect(this.kindDialogWindow).toBeVisible();
+    await this.dialogYesButton.click({ timeout: 5000 });
+    await this.dialogYesButton.click({ timeout: 5000 });
+    await this.dialogCloseButton.click({ timeout: 5000 });
```
Comment (cbr7) [inline tests/playwright/src/model/workbench/status-bar.ts]: Default timeout is 5 seconds, adding it manually does nothing.

File: tests/playwright/src/model/workbench/status-bar.ts [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+  }
+
+  public getKindInstallationButton(): Locator {
+    return this.kindInstallationButton;
```
Comment (cbr7) [inline tests/playwright/src/model/workbench/status-bar.ts]: What is the point of this method? We can directly access the locator field that it returns if it's specifically needed.

File: tests/playwright/src/specs/kind.spec.ts [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+});
+
+afterAll(async () => {
+  await pdRunner.close();
+});
+
+describe('Kind End-to-End Tests', async () => {
+  test.skipIf(isCI && isLinux)('Install Kind CLI if Not Present', async () => {
+    await navigationBar.openSettings();
+    if ((await resourcePage.ResourceExists('kind').isVisible({ timeout: 5000 })) === false && !isLinux) {
```
Comment (cbr7) [inline tests/playwright/src/specs/kind.spec.ts]: isVisible() methods from PW does not benefit from timeout, it will always return instantly the value as it's currently in the system, it has no implicit waiter baked in. Also I'm not at all sure I understand why you are checking equality agaisnt false && !isLinux given that the test is skipped if it is linux.

File: tests/playwright/src/specs/kind.spec.ts [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+      await statusBar.installKindCLI();
```
Comment (cbr7) [inline tests/playwright/src/specs/kind.spec.ts]: There's no waiter or validation for the installation from what I see,

File: tests/playwright/src/specs/kind.spec.ts [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+    }
+  });
+  test('Verify Kind CLI Installation', async () => {
+    await navigationBar.openSettings();
+    await playExpect(statusBar.getKindInstallationButton()).not.toBeVisible({ timeout: 5000 });
+    await playExpect(resourcePage.ResourceExists('kind')).toBeVisible({ timeout: 5000 });
```
Comment (cbr7) [inline tests/playwright/src/specs/kind.spec.ts]: Default timeout is 5000 ms, so there is no point to specifying it as such.

File: tests/playwright/src/specs/kind.spec.ts [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+  });
+  test('Explore and manage Kind extension lifecycle', async () => {
+    const extensionsPage = new ExtensionsPage(page);
+    await navigationBar.openExtensions();
+    await playExpect.poll(async () => await extensionsPage.extensionIsInstalled(extensionLabel)).toBeTruthy();
+    await playExpect(extensionsPage.getExtension(extensionLabel).getByText('ACTIVE')).toBeVisible();
```
Comment (cbr7) [inline tests/playwright/src/specs/kind.spec.ts]: There is a method in the POM that returns the status of an extension, you do not need to manually grab the status field in the test, in general handling the web elements themselves inside the actual tests should be avoided as much as possible since it defeats the purpose of using POM

File: tests/playwright/src/specs/kind.spec.ts [hunk 8]
```diff
@@ -0,0 +1,86 @@
...
+    await extensionsPage.getExtension(extensionLabel).getByRole('button', { name: 'Stop' }).click();
```
Comment (cbr7) [inline tests/playwright/src/specs/kind.spec.ts]: This should be encapsulated and exposed as a method from POM, not handled directly in the test.

File: tests/playwright/src/specs/kind.spec.ts [hunk 9]
```diff
@@ -0,0 +1,86 @@
...
+    await playExpect(extensionsPage.getExtension(extensionLabel).getByText('DISABLED')).toBeVisible();
```
Comment (cbr7) [inline tests/playwright/src/specs/kind.spec.ts]: Same as above.

File: tests/playwright/src/specs/kind.spec.ts [hunk 10]
```diff
@@ -0,0 +1,86 @@
...
+    await navigationBar.openSettings();
+    await playExpect(resourcePage.ResourceExists('kind')).not.toBeVisible({ timeout: 5000 });
+    await navigationBar.openExtensions();
+    await extensionsPage.getExtension(extensionLabel).getByRole('button', { name: 'Start' }).click();
```
Comment (cbr7) [inline tests/playwright/src/specs/kind.spec.ts]: Same as above for the Stop button,.

File: tests/playwright/src/specs/kind.spec.ts [hunk 11]
```diff
@@ -0,0 +1,86 @@
...
+  });
+  test('Explore and manage Kind extension lifecycle', async () => {
+    const extensionsPage = new ExtensionsPage(page);
+    await navigationBar.openExtensions();
+    await playExpect.poll(async () => await extensionsPage.extensionIsInstalled(extensionLabel)).toBeTruthy();
+    await playExpect(extensionsPage.getExtension(extensionLabel).getByText('ACTIVE')).toBeVisible();
+    await extensionsPage.getExtension(extensionLabel).getByRole('button', { name: 'Stop' }).click();
+    await playExpect(extensionsPage.getExtension(extensionLabel).getByText('DISABLED')).toBeVisible();
+    await navigationBar.openSettings();
+    await playExpect(resourcePage.ResourceExists('kind')).not.toBeVisible({ timeout: 5000 });
```
Comment (cbr7) [inline tests/playwright/src/specs/kind.spec.ts]: Please see previous comment about ResourceExists() and how it should be refactored, it should return true or false

File: tests/playwright/src/specs/kind.spec.ts [hunk 12]
```diff
@@ -0,0 +1,86 @@
...
+    await navigationBar.openExtensions();
+    await extensionsPage.getExtension(extensionLabel).getByRole('button', { name: 'Start' }).click();
+    await navigationBar.openSettings();
+    await playExpect(resourcePage.ResourceExists('kind')).toBeVisible({ timeout: 5000 });
```
Comment (cbr7) [inline tests/playwright/src/specs/kind.spec.ts]: Same as above.

**Review by cbr7 — CHANGES_REQUESTED** [review]
Left some comments.

File: package.json [hunk 13]
```diff
@@ -50,7 +50,7 @@
     "test:unit": "npm run test:main && npm run test:preload && npm run test:preload-docker-extension && npm run test:preload-webview && npm run test:ui && npm run test:renderer && npm run test:scripts:stylesheet && npm run test:tools && npm run test:extensions",
     "test:e2e": "npm run test:e2e:build && npm run test:e2e:run",
     "test:e2e:build": "cross-env NODE_ENV=development MODE=development DEBUG=pw:browser npm run build",
-    "test:e2e:run": "xvfb-maybe --auto-servernum --server-args='-screen 0 1280x960x24' -- vitest run tests/playwright/src/specs/ --pool=threads --poolOptions.threads.singleThread --poolOptions.threads.isolate --no-file-parallelism",
+    "test:e2e:run": "xvfb-maybe --auto-servernum --server-args='-screen 0 1280x960x24' -- vitest run tests/playwright/src/specs/kind.spec.ts --pool=threads --poolOptions.threads.singleThread --poolOptions.threads.isolate --no-file-parallelism",
```
Comment (cbr7) [inline package.json]: Please revert this change, it will make it so only this test file would be executed.

**Review by cbr7 — CHANGES_REQUESTED** [review]

**Review by cbr7 — APPROVED** [review]

File: tests/playwright/src/utility/operations.ts [hunk 14]
```diff
@@ -155,6 +155,13 @@ export async function handleConfirmationDialog(
   await button.click();
 }
 
+export async function closeConfirmationDialog(page: Page, dialogTitle = 'Confirmation'): Promise<void> {
```
Comment (odockal) [inline tests/playwright/src/utility/operations.ts]: I think that you can improve `handleConfimationDialog` function so it take another two parameters (type string, first `confirmationButton` with default value "Yes" and `cancelButton` with default value `Cancel`) which would then work well for your kind installation dialogs.

File: tests/playwright/src/model/workbench/status-bar.ts [hunk 15]
```diff
@@ -0,0 +1,40 @@
...
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ***********************************************************************/
+
+import type { Locator, Page } from '@playwright/test';
+
+import { closeConfirmationDialog, handleConfirmationDialog } from '../../utility/operations';
+import { BasePage } from '../pages/base-page';
+
+export class StatusBar extends BasePage {
```
Comment (odockal) [inline tests/playwright/src/model/workbench/status-bar.ts:24]: You need to add also this class to the `index.ts` so it will be part of npm `@podman-desktop/tests-playwright` package.

**Review by odockal — CHANGES_REQUESTED** [review]
Do not forget to all add env. var into `e2e-test-main.yaml` e2e test job.

**Comment by odockal** [comment]
Workflow run: https://github.com/containers/podman-desktop/actions/runs/9889805221

**Review by odockal — APPROVED** [review]

**Review by odockal — APPROVED** [review]
